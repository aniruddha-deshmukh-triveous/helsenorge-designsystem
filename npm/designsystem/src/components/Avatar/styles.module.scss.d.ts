export type Styles = {
  avatar: string;
  'avatar--black': string;
  'avatar--selected': string;
};

export type ClassNames = keyof Styles;

declare const styles: Styles;

export default styles;
