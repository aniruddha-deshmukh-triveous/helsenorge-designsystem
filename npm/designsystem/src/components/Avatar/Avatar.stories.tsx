import React from 'react';

import { ComponentStory, ComponentMeta } from '@storybook/react';

import Avatar from './Avatar';

export default {
  title: 'Avatar',
  component: Avatar,
  argTypes: {
    children: {
      control: 'text',
      defaultValue: 'Line Danser',
    },
    selected: {
      control: 'boolean',
      defaultValue: false,
    },
  },
} as ComponentMeta<typeof Avatar>;

export const Default: ComponentStory<typeof Avatar> = (args: any) => <Avatar {...args}>{args.children}</Avatar>;

export const Inverted: ComponentStory<typeof Avatar> = (args: any) => (
  <Avatar {...args} variant={'black'}>
    {args.children}
  </Avatar>
);
