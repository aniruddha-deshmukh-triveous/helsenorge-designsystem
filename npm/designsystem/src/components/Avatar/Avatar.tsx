import React from 'react';
import Icon from '../Icons';
import Check from '../Icons/Check';
import cn from 'classnames';

import styles from './styles.module.scss';

interface AvatarProps {
  children: string;
  selected?: boolean;
  className?: string;
}

const Avatar = React.forwardRef(function AvatarForwardedRef(props: AvatarProps, ref: React.ForwardedRef<HTMLElement>) {
  const { children, className = '', selected = false } = props;
  const truncatedName = children.charAt(0).toLocaleUpperCase() + children.substring(1, 2);
  return (
    <span className={cn(styles.avatar, selected && styles['avatar--selected'], className)} ref={ref}>
      {selected ? <Icon svgIcon={Check} size={48} color="kiwi" /> : truncatedName}
    </span>
  );
});

export default Avatar;
