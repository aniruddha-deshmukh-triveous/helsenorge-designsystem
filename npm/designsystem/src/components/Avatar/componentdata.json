{
  "props": {
    "children": {
      "defaultValue": null,
      "description": "Name to display in the avatar. Will be truncated to the first two characters.",
      "name": "children",
      "parent": {
        "fileName": "src/components/Avatar/Avatar.tsx",
        "name": "AvatarProps"
      },
      "declarations": [
        {
          "fileName": "src/components/Avatar/Avatar.tsx",
          "name": "AvatarProps"
        }
      ],
      "required": true,
      "type": {
        "name": "string"
      }
    },
    "selected": {
      "defaultValue": null,
      "description": "Displays a check icon to indicated the selected state.",
      "name": "selected",
      "parent": {
        "fileName": "src/components/Avatar/Avatar.tsx",
        "name": "AvatarProps"
      },
      "declarations": [
        {
          "fileName": "src/components/Avatar/Avatar.tsx",
          "name": "AvatarProps"
        }
      ],
      "required": false,
      "type": {
        "name": "boolean"
      }
    },
    "className": {
      "defaultValue": null,
      "description": "Adds custom classes to the element.",
      "name": "className",
      "parent": {
        "fileName": "src/components/Avatar/Avatar.tsx",
        "name": "AvatarProps"
      },
      "declarations": [
        {
          "fileName": "src/components/Avatar/Avatar.tsx",
          "name": "AvatarProps"
        }
      ],
      "required": false,
      "type": {
        "name": "string"
      }
    }
  }
}
