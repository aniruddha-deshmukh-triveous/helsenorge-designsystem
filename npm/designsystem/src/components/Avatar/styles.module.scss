@import '../../scss/_palette.scss';

.avatar {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 3rem;
  height: 3rem;
  font-size: 1.5rem;
  font-weight: 600;
  background-color: $blueberry500;
  color: $white;
  &--selected {
    background-color: transparent;
  }
  &--black {
    background-color: $neutral900;
  }
  &--selected#{&}--black {
    background-color: transparent;
  }
}
