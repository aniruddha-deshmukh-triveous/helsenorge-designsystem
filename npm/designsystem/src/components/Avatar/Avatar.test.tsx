import React from 'react';
import { screen, render } from '@testing-library/react';
import Avatar from './Avatar';

describe('Gitt at Avatar skal vises', (): void => {
  describe('N책r selected er false', (): void => {
    it('S책 vises teksten i Avatar-komponenten', (): void => {
      const { container } = render(<Avatar>Line Danser</Avatar>);

      const text = screen.getByText('Li');
      expect(text).toBeVisible();
      expect(container).toMatchSnapshot();
    });
  });
  describe('N책r selected er true', (): void => {
    it('S책 vises ikke teksten', (): void => {
      const { container } = render(<Avatar selected>Line Danser</Avatar>);

      const text = screen.queryByText('Li');
      expect(text).not.toBeInTheDocument();
      expect(container).toMatchSnapshot();
    });
  });
});
