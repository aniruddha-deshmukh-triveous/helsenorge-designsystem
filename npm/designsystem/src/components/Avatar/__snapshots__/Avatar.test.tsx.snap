// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gitt at Avatar skal vises Når selected er false Så vises teksten i Avatar-komponenten 1`] = `
<div>
  <span
    class="avatar"
  >
    Li
  </span>
</div>
`;

exports[`Gitt at Avatar skal vises Når selected er true Så vises ikke teksten 1`] = `
<div>
  <span
    class="avatar avatar--selected"
  >
    <svg
      aria-hidden="true"
      class="sc-AxjAm jPyZsV hnds-style-icon "
      fill="#126F87"
      height="48"
      role="img"
      viewBox="0 0 48 48"
      width="48"
    >
      <path
        d="M22.504 31.198l-9.59-9.966 1.441-1.387 8.149 8.468 14.455-15.016 1.441 1.386z"
      />
    </svg>
  </span>
</div>
`;

exports[`Gitt at Avatar skal vises Når variant er black Så skal komponenten være svart 1`] = `
<div>
  <span
    class="avatar avatar--black"
  >
    Ba
  </span>
</div>
`;

exports[`Gitt at Avatar skal vises Når variant er black og selected Så skal bakgrunnen være transparent og fargen på ikon er svart 1`] = `
<div>
  <span
    class="avatar avatar--selected avatar--black"
  >
    <svg
      aria-hidden="true"
      class="sc-AxjAm jPyZsV hnds-style-icon "
      fill="#2B2C2B"
      height="48"
      role="img"
      viewBox="0 0 48 48"
      width="48"
    >
      <path
        d="M22.504 31.198l-9.59-9.966 1.441-1.387 8.149 8.468 14.455-15.016 1.441 1.386z"
      />
    </svg>
  </span>
</div>
`;
